### DISCUSSION TITLE
ImageMagick not rendering images correctly since recent security update

### DISCUSSION THREAD

**Discussion by paladox** [ImageMagick/ImageMagick#2259] [Help]
Hi, Since we updated imagemagik from 8:6.9.10.23+dfsg-2.1 to 8:6.9.10.23+dfsg-2.1+deb10u1 in debain buster, we've had issues with image rendering. Could you advise on any steps to assist in debugging or provide some assistance?

**Comment by snibgo** [comment]
Please be specific, and provide a simple command that doesn't work. If it needs any input files, please supply them.

**Comment by fmw42** [comment]
Please show what you get from

`convert -version`

**Comment by paladox** [comment]
```
root@mw4:/home/paladox# convert -version
Version: ImageMagick 6.9.10-23 Q16 x86_64 20190101 https://imagemagick.org
Copyright: © 1999-2019 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP 
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png tiff webp wmf x xml zlib
```

This is the command i see MediaWiki run:

```
2020-07-12 20:11:13 test2 test2wiki: MediaWiki\Shell\Command::execute: /bin/bash '/srv/mediawiki/w/includes/shell/limit.sh' 'LANG='\''en'\'' convert -background "#ffffff00" -thumbnail 10x10\! '\''/tmp/svg_dbf77aef1382e2d0c174b4c8/Miraheze-Logo-6.svg'\'' PNG:'\''/tmp/transform_eee79e3f05d4.png'\''' 'MW_INCLUDE_STDERR=1;MW_CPU_LIMIT=180; MW_CGROUP='\'''\''; MW_MEM_LIMIT=2097152; MW_FILE_SIZE_LIMIT=102400; MW_WALL_CLOCK_LIMIT=180; MW_USE_LOG_PIPE=yes'
```

And the file is https://commons.wikimedia.org/wiki/File:Miraheze-Logo.svg, it basically ends up looking like https://test2.miraheze.org/wiki/File:Miraheze-Logo-6.svg

This is because some how it's doing 10px.

**Reply by fmw42** [reply]
Your command has `-thumbnail 10x10\!`, which will make the PNG output 10x10 pixels. If that is not what you want, then you need to find out why MediaWiki is creating that command. It is not ImageMagick error, but an error in your MediaWiki as best I can tell. Or an error in whomever created the convert command that is being run.

Perhaps you wanted 10x10%?  If so, the -thumbnail command needs to have the % added.

**Comment by snibgo** [comment]
Your command calls a script. This seems to rasterize a SVG, then resize that to exactly 10x10 pixels, using "-thumbnail". 

> This is because some how it's doing 10px.
Yes. If that isn't what you want, then don't resize to 10x10.

You might also want to set "-density 300" or something, before reading the SVG.

Also note that rasterizing a SVG is delegated to either Inkscape, or internal RSVG, or internal MSVG. This may have changed in your upgrade.
